### DISCUSSION TITLE
Use dynamically generated types in composables and components exported by a module

### DISCUSSION THREAD

**Discussion by Rigo-m** [nuxt/nuxt#21208] [Questions]
Reproduction: https://github.com/Rigo-m/nuxt-module-types-test

What I'm trying to achieve is to have composables and components auto imported, exposed from a module that uses types that gets generated by the module itself after it's been installed.

I've tried replicating some strategies that I saw on some other modules but I didn't find any luck whatsoever.
@danielroe I think you might have a clear idea on how to tackle this kind of things, thanks in advance for any help to anyone who can provide some

**Comment by danielroe** [comment]
Linking: https://github.com/unjs/mkdist/pull/154 and https://stackblitz.com/edit/github-kafibb.

**Reply by Rigo-m** [reply]
Can confirm that by using declare type the component gets compiled correctly, while the composable d.ts gets compiled like this:
```ts
declare const useTest: <KType extends string | number | symbol>(key: KType) => void;
```
